src/main/java/com/snowflake/kafka/connector/internal/streaming/TopicPartitionChannel.java

```diff
@@ -992,23 +993,28 @@ protected long getApproxSizeOfRecordInBytes(SinkRecord kafkaSinkRecord) {
       return 0L;
     }
 
-    // get the row that we want to insert into Snowflake.
-    Map<String, Object> tableRow =
-        recordService.getProcessedRecordForStreamingIngest(snowflakeRecord);
-    // need to loop through the map and get the object node
-    for (Map.Entry<String, Object> entry : tableRow.entrySet()) {
-      sinkRecordBufferSizeInBytes += entry.getKey().length() * 2L;
-      // Can Typecast into string because value is JSON
-      Object value = entry.getValue();
-      if (value instanceof String) {
-        sinkRecordBufferSizeInBytes += ((String) value).length() * 2L; // 1 char = 2 bytes
-      } else {
-        // for now it could only be a list of string
-        for (String s : (List<String>) value) {
-          sinkRecordBufferSizeInBytes += s.length() * 2L;
+    try {
+      // get the row that we want to insert into Snowflake.
+      Map<String, Object> tableRow =
+          recordService.getProcessedRecordForStreamingIngest(snowflakeRecord);
+      // need to loop through the map and get the object node
+      for (Map.Entry<String, Object> entry : tableRow.entrySet()) {
+        sinkRecordBufferSizeInBytes += entry.getKey().length() * 2L;
+        // Can Typecast into string because value is JSON
+        Object value = entry.getValue();
+        if (value instanceof String) {
+          sinkRecordBufferSizeInBytes += ((String) value).length() * 2L; // 1 char = 2 bytes
+        } else {
+          // for now it could only be a list of string
+          for (String s : (List<String>) value) {
+            sinkRecordBufferSizeInBytes += s.length() * 2L;
+          }
         }
       }
+    } catch (JsonProcessingException e) {
+      // We ignore any errors here because this is just calculating the record size
```

Comment:
Should we instead explicitly handle this or log the exception? What happens if we continue?

Reply:
I feel its okay to skip this and continue since we dont want to calculate/add this record's byte length to memory. A warn should be useful here below in catch exception.

Reply:
We will call `getProcessedRecordForStreamingIngest` in two places and I added the log in the other place, also I create https://snowflakecomputing.atlassian.net/browse/SNOW-663981 since it's not ideal to process every record twice, ideally we should process every record only once and save that